src/Symfony/Component/Security/Http/Authenticator/AccessTokenAuthenticator.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Http\AccessToken\AccessTokenExtractorInterface;
+use Symfony\Component\Security\Http\AccessToken\AccessTokenHandlerInterface;
+use Symfony\Component\Security\Http\AccessToken\DefaultAuthenticationFailureHandler;
+use Symfony\Component\Security\Http\Authentication\AuthenticationFailureHandlerInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\Credentials\AccessTokenCredentials;
+use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
+use Symfony\Component\Security\Http\Authenticator\Token\PostAuthenticationToken;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Provides an implementation of the RFC6750 of an authentication via
+ * an access token.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ */
+class AccessTokenAuthenticator implements AuthenticatorInterface
+{
+    private AuthenticationFailureHandlerInterface $failureHandler;
+
+    public function __construct(
+        private readonly UserProviderInterface $userProvider,
+        private readonly AccessTokenHandlerInterface $accessTokenHandler,
+        private readonly AccessTokenExtractorInterface $accessTokenExtractor,
+        private readonly ?AuthenticationSuccessHandlerInterface $successHandler = null,
+        ?AuthenticationFailureHandlerInterface $failureHandler = null,
+        string $realm = null,
+    ) {
+        $this->failureHandler = $failureHandler ?? new DefaultAuthenticationFailureHandler($realm);
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return null !== $this->accessTokenExtractor->extractAccessToken($request);
```

Comment:
Can we let it be as lazy as possible? cc @wouterj 
```php
return null === $this->accessTokenExtractor->extractAccessToken($request) ? null : true;
```

Reply:
I am not sure tu clearly understand the difference 🤔.

Reply:
Related to https://symfony.com/blog/new-in-symfony-4-4-lazy-firewalls. 
Basically, returning `null` instead of `false` from `AuthenticatorInterface::supports()` allows the system to avoid calling `AuthenticatorInterface::authenticate()` unless the token storage is actually used (either by the authorization layer or an explicit call to `getToken()`). This allows not starting the session on stateful firewalls until it's really needed, making the response cacheable in case it's not accessed at all.

Reply:
OK that's clear now. Change is done